### PR TITLE
add workflow to create dependabot pr

### PR DESCRIPTION
**Pull request by codeboten**
This pr adds a workflow to replace the manual work required to commit all the dependabot PRs. The workflow:
- checks out the code
- runs .github/workflows/scripts/dependabot-pr.sh which looks for all dependabot PRs and applies the dependency update
- runs tidy/builds the collector
- commits a change and creates a PR

The workflow is triggered manually and the PR will be opened by the github-actions bot

### CODE REVIEW

File: .github/workflows/scripts/dependabot-pr.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#!/bin/bash -ex
+
+git config user.name $GITHUB_ACTOR
+git config user.email $GITHUB_ACTOR@users.noreply.github.com
```
Comment (TylerHelmuth) [inline .github/workflows/scripts/dependabot-pr.sh:4]: Will this handle the situation where the workflow starter is using the "Keep my email addresses private" option?  For example, my `users.noreply.github.com` email is `12352919+TylerHelmuth@users.noreply.github.com`.  Will `GITHUB_ACTOR` contain `TylerHelmuth` or `12352919+TylerHelmuth`?  Or does it not matter?

File: .github/workflows/scripts/dependabot-pr.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+requests=$(gh pr list --search "author:app/dependabot" --json number,title --template '{{range .}}{{tablerow .title}}{{end}}')
+message=""
+
+for line in $requests; do
+    if [[ $line != Bump* ]]; then 
+        continue
+    fi
+
+    module=$(echo $line | cut -f 2 -d " ")
+    if [[ $module == go.opentelemetry.io/collecto* ]]; then
```
Comment (TylerHelmuth) [inline .github/workflows/scripts/dependabot-pr.sh:19]: which packages are covered by `go.opentelemetry.io/collecto*` but not by `go.opentelemetry.io/collector*`?

**Review by TylerHelmuth — COMMENTED** [review]
Is this a script you've been running locally?  Have you been able to test it as a github action?

**Comment by codeboten** [comment]
> Is this a script you've been running locally? Have you been able to test it as a github action?

I've run it in my fork, you can see the resulting PR here: https://github.com/codeboten/opentelemetry-collector-contrib/pull/2748

Here's the action being run: https://github.com/codeboten/opentelemetry-collector-contrib/runs/6802784258?check_suite_focus=true

File: .github/workflows/scripts/dependabot-pr.sh (hunk 1 above)
Comment (codeboten) [reply .github/workflows/scripts/dependabot-pr.sh:4]: i was mistaken, the `GITHUB_ACTOR` is the github actions bot

Comment (TylerHelmuth) [reply .github/workflows/scripts/dependabot-pr.sh:4]: In the workflow you linked it looks like it is the person who triggered the workflow:

image

Comment (codeboten) [reply .github/workflows/scripts/dependabot-pr.sh:4]: Right, sorry this is confusing. The GITHUB_ACTOR is the user that triggered the action, which is used to create the commit, but the PR is using the GITHUB_TOKEN which is the github action bot's token
Screen Shot 2022-06-09 at 8 57 03 AM
Screen Shot 2022-06-09 at 8 57 12 AM

Comment (codeboten) [reply .github/workflows/scripts/dependabot-pr.sh:4]: Either ways it doesn't look like the email matters to github

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**